---
date_created: Monday, April 22nd 2024, 10:58:22 pm
date_updated: Wednesday, January 22nd 2025, 11:54:06 pm
title: C++异常
author: hacket
categories:
  - C&C++
category: CPP
tags: [CPP]
toc: true
description: 
dg-publish: true
dg-enable-search: true
dg-show-local-graph: true
dg-show-toc: true
dg-show-file-tree: true
image-auto-upload: true
feed: show
format: list
date created: 2024-04-11 00:16
date updated: 2024-12-27 23:51
aliases: [C++ 异常]
linter-yaml-title-alias: C++ 异常
---

# C++ 异常

## C++ 异常

1. `throw` 抛出异常
2. `try{}catch()` 捕获异常
3. 可以抛出任何对象

```c
void test1()
{
	throw "测试!";
}

void test2()
{
	throw exception("测试");
}

try {
	test1();
}
catch (const char *m) {
	cout << m << endl;
}
try {
	test2();
}
catch (exception  &e) {
	cout << e.what() << endl;
}

//自定义
class MyException : public exception
{
public:
   virtual char const* what() const
    {
        return "myexception";
    }
};

//随便抛出一个对象都可以
```
